lib/purescript-cst/src/Language/PureScript/CST/Convert.hs

```diff
@@ -540,13 +541,16 @@ convertDeclaration fileName decl = case decl of
     where
       -- truncate to 25 chars to reduce verbosity
       -- of name and still keep it readable
-      -- unique identifier will be appended to this name
-      -- in desugaring proces
+      -- name will be used to create a GenIdent
+      -- in desugaring process
       genName :: Text.Text
-      genName = "$_" <> Text.take 25 (className <> typeArgs) <> "_"
+      genName = Text.take 25 (className <> typeArgs)
```

Comment:
Why is the final `_` character being removed? Is this just preference? 

I'm not for or against this as I think the larger readability problems are addressed in this PR. To me, this seems more like a 'stylistic' change than anything else.

Reply:
It is cosmetic, yes. But note that the renamer only appends numbers to the end of a `GenIdent` if necessary to disambiguate. So with the `_` character, instances were getting named `classNameArg_`, which is why I removed it.

Reply:
Ah, makes sense.